js/src/jupyter-leaflet.js

```diff
@@ -113,11 +114,20 @@ var LeafletUILayerView = LeafletLayerView.extend({
 
 var LeafletMarkerView = LeafletUILayerView.extend({
 
+    initialize: function() {
+        // Public constructor
+        LeafletMarkerView.__super__.initialize.apply(this, arguments);
+        this.icon_promise = Promise.resolve();
+    },
+
+
     create_obj: function () {
         var that = this;
+        var options = this.get_options();
+        delete options['icon'];
         this.obj = L.marker(
             this.model.get('location'),
-            this.get_options()
+            options
```

Comment:
If you remove `o=True` from the icon attr in the Python model you can undo those changes.